open-hackathon-server/src/hackathon/hack/host_server_manager.py

```diff
@@ -49,7 +72,432 @@ def get_available_docker_host(self, req_count, hackathon):
+        """
+        container_name = self.util.safe_get_config('dockerhostserver.azure.container', 'dockerhostprivatecontainer')
+        sms = self.__get_sms_object(hackathon_id)
+        storage_accounts = sms.list_storage_accounts()
+        # check storage account one by one, return True once find a qualified one
+        for storage in storage_accounts.storage_services:
+            try:
+                storage_response = sms.get_storage_account_keys(storage.service_name)
+            except Exception as e:
+                self.log.error('Encounter an error when checking storage_account:%s ' % storage.service_name)
+                self.log.error(e)
+                continue
+            blob_service = BlobService(account_name=storage.service_name,
+                                       account_key=storage_response.storage_service_keys.primary,
+                                       host_base=self.util.safe_get_config('dockerhostserver.storage.host_base',
+                                                                           '.blob.core.chinacloudapi.cn'))
+            try:
+                blob_service.get_container_metadata(container_name)
+                return True, storage.service_name, container_name
+            except Exception as e:
+                if e.message != 'Not found (The specified container does not exist.)':
+                    self.log.error('Encounter an error when checking container:%s ' % container_name)
+                    self.log.error(e)
+                    continue
+            try:
+                blob_service.create_container(container_name=container_name, x_ms_blob_public_access='container')
+                return True, storage.service_name, container_name
+            except Exception as e:
+                self.log.error('Encounter an error when creating container:%s ' % container_name)
+                self.log.error(e)
+        return False, None, None
+
+    def __get_available_cloud_service(self, hackathon_id):
+        """
+        Get available cloud service
+
+        :param hackathon_id: the id of hackathon
+        :type hackathon_id: integer
+
+        :return: if there is a available cloud service, then return (True, cloud service name). Otherwise, return
+                (False, None).
+        :rtype: 2-element tuple: (bool, str|unicode)
+        """
+        config_name = self.util.safe_get_config('dockerhostserver.azure.cloud_service.name',
+                                                'dockerhostprivatecloudservice')
+        service_name = config_name + str(hackathon_id)
+        service_location = self.util.safe_get_config('dockerhostserver.azure.cloud_service.location', 'China East')
+        sms = self.__get_sms_object(hackathon_id)
+        try:
+            sms.get_hosted_service_properties(service_name, True)
+            return True, service_name
+        except Exception as e:
+            if e.message != 'Not found (Not Found)':
```

Comment:
Do we have to use `e.message` for judgement? Is there any other option such as `status`, `state` or `code`?